vortex-expr/src/select.rs

```diff
@@ -0,0 +1,85 @@
+use std::any::Any;
+use std::collections::HashSet;
+
+use vortex::array::StructArray;
+use vortex::variants::StructArrayTrait;
+use vortex::{Array, IntoArray};
+use vortex_dtype::field::Field;
+use vortex_error::{vortex_bail, VortexResult};
+
+use crate::{unbox_any, Column, VortexExpr};
+
+// TODO(robert): Support wildcard and column exclusions
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Select {
+    columns: Vec<Column>,
+}
+
+impl Select {
+    pub fn new(columns: Vec<Column>) -> Self {
+        Self { columns }
+    }
+}
+
+impl PartialEq<dyn Any> for Select {
```

Comment:
nit: `impl PartialEq<dyn Any>` is below `impl VortexExpr` for all the other expression types